### ISSUE TITLE
`throw-new-error` false positive

### ISSUE THREAD

**Issue by brokenmass** [sindresorhus/eslint-plugin-unicorn#2360]
after last release throw new error reports any call to function that are PascalCase and contains the word “Error” as invalid.

Example:

‘’’
const ParseError = () => 0;

// this is reported as invalid
cont result = ParseError();
‘’’

In my specific case I’m getting the error when using a class decorator to track custom error classes

‘’’
@RegisterServiceError()
class  SomeError extends BaseError {}
’’’

easiest solution would be to at least allow a list of name exceptions to the rule.

**Comment by fregante** [comment]
It's not quite clear why that should be valid. Only classes and JSX functions should be in pascal case

**Comment by brokenmass** [comment]
Using PascalCase only for classes and components is a naming convention, not a js requirement. It would be like this plugin giving false positive if the line is indented with tabs instead of white spaces.

**Comment by fregante** [comment]
> is a naming convention

This rule is specifically about enforcing that naming convention. If you don't agree with the convention, the rule can be disabled.

Specifically, I don’t think your request is compatible with the rule. 

Also I can't reproduce the specific issue with either one of your code examples. Please paste a real example with the output of ESLint for that line

**Comment by fregante** [comment]
> is a naming convention

This rule is specifically about enforcing that naming convention. If you don't agree with the convention, the rule can be disabled.



> ```js
> const ParseError = () => 0;
> const result = ParseError();
> ```

This is exactly what the rule prevents. If you want to use that, you can disable the rule.


> Using PascalCase for decorators (especially decorators factories) is pretty much standard.

If the rule is triggered on `@RegisterServiceError()`, then it's a bug, I agree. Here's a full repro that triggers both `new-cap` and `unicorn/throw-new-error`

```js
function RegisterServiceError() {
	return function <T extends new (...arguments_: any[]) => Error>(constructor: T) {
		return constructor;
	};
}

@RegisterServiceError()
export class SomeError extends Error {}
```

**Comment by CatsMiaow** [comment]
Same issue
<img width="733" alt="image" src="https://github.com/user-attachments/assets/98aa2534-d4f4-41fc-99c9-8b3dafa48773">

> https://github.com/nestjs/bull/blob/0e1d03cf3e1c8ce9dac7197c0631647f79cc7164/packages/bull/lib/decorators/queue-hooks.decorators.ts#L20
